app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,64 @@
+class RecipesController < ApplicationController
+  load_and_authorize_resource
+  before_action :authenticate_user!
+  before_action :find_user
+
+  def index
+    @recipes = current_user.recipes
+  end
+
+  def show
+    @recipe = Recipe.find(params[:id])
+    @recipe_foods = @recipe.recipe_foods.includes(:food)
+  end
+
+  def new
+    @recipe = Recipe.new
+  end
+
+  def create
+    @recipe = Recipe.new(recipe_params)
+    @recipe.user = @user
+    if @recipe.save
+      redirect_to recipe_path(id: @recipe.id), notice: 'Recipe created successfully'
+    else
+      flash.now[:alert] = @recipe.errors.full_messages.first if @recipe.errors.any?
```

Comment:
- [Optional ]  You did great to modify the design, but I am getting an issue while I am creating a recipe that the page is broken after clicking the add recipe button. please fix is so that After adding recipe I am taken to the list of the recipe I have.  
![image](https://github.com/Itswali/Recipe-App/assets/43877689/52167aeb-b61c-495f-977a-78e374abaa9e)